### PR TITLE
move some error and warning tests to NUnit

### PR DESCRIPTION
**Pull request by smcl**
In this PR I've moved the following tests over to the NUnit:

* tests/fsharpqa/Source/ErrorMessages/NameResolution/E_FieldNotInRecord.fs
* tests/fsharpqa/Source/ErrorMessages/NameResolution/E_RecordFieldProposal.fs
* tests/fsharpqa/Source/ErrorMessages/NameResolution/E_GlobalQualifierAfterDot.fs
* tests/fsharpqa/Source/ErrorMessages/Warnings/WarnIfPossibleAssignment.fs
* tests/fsharpqa/Source/ErrorMessages/Warnings/WarnIfPossibleAssignmentToMutable.fs
* tests/fsharpqa/Source/ErrorMessages/Warnings/WarnIfPossibleDotNetPropertySetter.fs
* tests/fsharpqa/Source/ErrorMessages/Warnings/WarnIfPossiblePropertySetter.fs
* tests/fsharpqa/Source/Warnings/DontWarnIfPropertyWithoutSetter.fs

The "E_GlobalQualifierAfterDot.fs" test (now `Global qualifier after dot`) raised a parse error so it wasn't possible to use the existing checks available in `CompilerAssert` so I created `CompilerAssert.ParseWithErrors` which is similar to `CompilerAssert .TypeCheckWithErrors`.

### CODE REVIEW

File: tests/fsharp/Compiler/CompilerAssert.fs [hunk 1]
```diff
@@ -217,3 +217,23 @@ module CompilerAssert =
                     Assert.AreEqual(expectedErrorMessage, errorMessage)
                 )
         
+    let ParseWithErrors (source: string) expectedParseErrors = 
+        lock gate <| fun () ->
+            let parseResults, _ = checker.ParseAndCheckFileInProject("test.fs", 0, SourceText.ofString source, defaultProjectOptions) |> Async.RunSynchronously
```
Comment (cartermp) [inline tests/fsharp/Compiler/CompilerAssert.fs]: Is it possible to use `ParseFile` (https://github.com/dotnet/fsharp/blob/master/src/fsharp/service/service.fsi#L113) instead?

File: tests/fsharp/Compiler/CompilerAssert.fs [hunk 2]
```diff
@@ -217,3 +217,23 @@ module CompilerAssert =
                     Assert.AreEqual(expectedErrorMessage, errorMessage)
                 )
         
+    let ParseWithErrors (source: string) expectedParseErrors = 
+        lock gate <| fun () ->
```
Comment (cartermp) [inline tests/fsharp/Compiler/CompilerAssert.fs]: If you use `ParseFile` then this may not be necessary, since all parsing operations are thread-safe

**Comment by KevinRansom** [comment]
@smcl, it looks as if some of these tests have already been migrated.

Although the other PR didn't migrate this test, so if you resolve the merge issues we can still take it :-)
````
    [<Test>]
    let ``Global qualifier after dot``() =
        CompilerAssert.ParseWithErrors
            """
let x = global.System.String.Empty.global.System.String.Empty
exit 0
            """
            [| FSharpErrorSeverity.Error, 1126, (2, 36, 2, 42), "'global' may only be used as the first name in a qualified path" |]
````




Thanks for doing this,

Kevin

File: tests/fsharp/Compiler/CompilerAssert.fs (hunk 1 above)
Comment (cartermp) [reply tests/fsharp/Compiler/CompilerAssert.fs]: Just confirming that I think `ParseFile` should be sufficient here, and you could probably get rid of the lock as a result

File: tests/fsharp/Compiler/CompilerAssert.fs (hunk 2 above)
Comment (smcl) [reply tests/fsharp/Compiler/CompilerAssert.fs]: Yep, I just got back and tried it out (with ``FSharpParsingOptions.Default`` as the fsharp parser opts, which I hope is OK). I'll back out the test dupes @KevinRansom's comment and sort the merge issues and hopefully all will be grand

File: tests/fsharp/Compiler/CompilerAssert.fs (hunk 1 above)
Comment (TIHan) [reply tests/fsharp/Compiler/CompilerAssert.fs]: Agree with @cartermp , see if you can use ParseFile.

**Comment by cartermp** [comment]
Thanks!

**Pull request merged**